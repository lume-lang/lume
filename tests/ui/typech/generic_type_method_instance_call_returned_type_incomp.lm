struct A<T> {}

impl<T> A<T> {
    pub fn bar(self, val: T) -> T {
        return val;
    }
}

fn bar() {
    let a: A<Boolean> = A<Boolean> { };

    let _: Int32 = a.bar(false);
}
