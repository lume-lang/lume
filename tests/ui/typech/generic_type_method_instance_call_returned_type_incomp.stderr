× error[LM4001]: mismatched types
    ╭─[generic_type_method_instance_call_returned_type_incomp.lm:12:22]
 11 │
 12 │     let _: Int32 = a.bar(false);
    ∶                      ^^^^^^^^^^^ expected type Int32, but found type Boolean...
 13 │ }
    ∶
 11 │
 12 │     let _: Int32 = a.bar(false);
    ∶            ^^^^^ ...because of type defined here
 13 │ }
    ╰──
   help: expected type Int32
            found type Boolean